### DISCUSSION TITLE
Completed: Store UI Settings Permanently in the Database

### DISCUSSION THREAD

**Discussion by exxus** [paperless-ngx/paperless-ngx#300] [Feature Requests] [answered]
Dear paperless-ngx team,
thank you for reviving this nice piece of software. I use Paperless-ngx 1.6.0 with docker-compose and want to report on issue.
When change the Display language and the Date display in the setting tab, an reload the page, the settings get applied.
But after restarting docker or my computer they are reset to default. All important documents are stored persistently.
Is it possible to also store the settings persistently or to add them per option in  _docker-compose.yml_?

Best regards
exxus

**Comment by shamoon** [comment]
Right now settings are stored in the browser "local storage". This is definitely something that should be discussed to change (likely not all settings should be put to db but certainly some / many). But its not such a small endeavor and involves database migrations, API change etc. But I definitely would support this.

Edit: settings should persist between docker restarts etc. as long as you dont clear your browser storage on the app

**Reply by sense4t** [reply]
I really thought that was a bug, too. I'm using several clients to access my paperless-ngx and up to know i expected that these settings stick to the current account (that's maybe an inconsistency with a common 'user-settings' paradigm).

**Reply by shamoon** [reply]
Yea there's really no way to know. The saved views are saved to the DB too, so its just not obvious. We could put something in the UI to make it apparent but realistically would be better to properly implement this (but at this point most of the work is on the Django side of things so that really aint my side of the biz)

**Reply by exxus** [reply]
Thank you for the explanation.
I have by default my browser set to clear cookies and cache on restart.
The restart of the docker image coincided with the restart of the browser.
That's what confused me. Know that I know the problem, I can use a different browser for paperless-ngx where the cache is not cleared or add an exception for paperless-ngx.
In the short term I think a hint on the settings page would be useful.
I agree, that in the long term the setting should be saved to the DB. Then the user experience would stay the same even when accessing from different devices / browsers.

**Comment by asitemade4u** [comment]
Thanks for the answer and the revival of Paperless, which is awesome.
Yes, I agree with the idea settings should definitely be placed into DB.

**Comment by shamoon** [comment, marked as answer]
Hi all, Im happy to report this was implemented in #919 and should be included in our next release (no ETA on that just yet). This should allow us to do some helpful stuff in the future too. 🎉
